circuits/src/zk_circuits/valid_offline_fee_settlement.rs

```diff
@@ -354,3 +351,156 @@ where
+
+#[cfg(any(test, feature = "test_helpers"))]
+pub mod test_helpers {
+    use circuit_types::{
+        native_helpers::{
+            compute_wallet_share_commitment, compute_wallet_share_nullifier, encrypt_note,
+            note_commitment, reblind_wallet,
+        },
+        note::Note,
+        wallet::Wallet,
+        Amount,
+    };
+    use constants::Scalar;
+    use rand::{thread_rng, Rng, RngCore};
+
+    use crate::zk_circuits::test_helpers::{
+        create_multi_opening, create_wallet_shares, PROTOCOL_KEY,
+    };
+
+    use super::{ValidOfflineFeeSettlementStatement, ValidOfflineFeeSettlementWitness};
+
+    /// Create a witness and statement for the `VALID OFFLINE FEE SETTLEMENT`
+    /// circuit
+    pub fn create_witness_statement<
+        const MAX_BALANCES: usize,
+        const MAX_ORDERS: usize,
+        const MERKLE_HEIGHT: usize,
+    >(
+        wallet: &Wallet<MAX_BALANCES, MAX_ORDERS>,
+    ) -> (
+        ValidOfflineFeeSettlementStatement<MAX_BALANCES, MAX_ORDERS>,
+        ValidOfflineFeeSettlementWitness<MAX_BALANCES, MAX_ORDERS, MERKLE_HEIGHT>,
+    )
+    where
+        [(); MAX_BALANCES + MAX_ORDERS]: Sized,
+    {
+        let mut rng = thread_rng();
+        let mut wallet = wallet.clone();
+        let mut updated_wallet = wallet.clone();
+
+        // Choose a balance to settle the fee from
+        let is_protocol = rng.gen_bool(0.5);
+        let send_index = rng.gen_range(0..MAX_BALANCES);
+
+        let send_bal = &mut wallet.balances[send_index];
+        let updated_bal = &mut updated_wallet.balances[send_index];
+        let send_mint = send_bal.mint.clone();
+        let send_amt = Amount::from(rng.next_u64());
+
+        // Modify the balances and select the key to encrypt under
+        let key = if is_protocol {
+            send_bal.protocol_fee_balance = send_amt;
+            updated_bal.protocol_fee_balance = Amount::from(0u8);
+            *PROTOCOL_KEY
+        } else {
+            send_bal.relayer_fee_balance = send_amt;
+            updated_bal.relayer_fee_balance = Amount::from(0u8);
+            wallet.managing_cluster
+        };
+
+        // Create secret shares for the wallets
+        let (original_wallet_private_shares, original_wallet_public_shares) =
+            create_wallet_shares(&wallet);
+
+        let (updated_wallet_private_shares, updated_wallet_public_shares) =
+            reblind_wallet(&original_wallet_private_shares, &updated_wallet);
+
+        // Create an opening for the original wallet
+        let commitment = compute_wallet_share_commitment(
+            &original_wallet_public_shares,
+            &original_wallet_private_shares,
+        );
+        let (root, opening) = create_multi_opening(&[commitment]);
+        let nullifier = compute_wallet_share_nullifier(commitment, wallet.blinder);
+
+        // Create the note
+        let note = Note {
+            mint: send_mint,
+            amount: send_amt,
+            receiver: wallet.managing_cluster,
```

Comment:
this should be `key` so that it depends on `is_protocol` appropriately

Reply:
fixed in follow up